File: bot/engine/src/main/kotlin/admin/bot/llm/settings/azureopenai/AzureOpenAISetting.kt
/*
 * Copyright (C) 2017/2021 e-voyageurs technologies
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ai.tock.bot.admin.bot.llm.settings.azureopenai

import ai.tock.bot.admin.bot.llm.settings.LLMProvider
import ai.tock.bot.admin.bot.llm.settings.LLMSetting

data class AzureOpenAISetting(
    override val apiKey: String,
    override val model: String,
    override val temperature: String? = null,
    override val prompt: String? = null,
    val deploymentName: String,
    val privateEndpointBaseUrl: String,
    val apiVersion: String,
) : LLMSetting(LLMProvider.AzureOpenAIService, apiKey, model, temperature, prompt)

File: llm/orchestrator-server/src/main/python/app/src/app/models/llm/azureopenai/azureopenaisetting.py
#   Copyright (C) 2023 Credit Mutuel Arkea
#
#   Licensed under the Apache License, Version 2.0 (the "License");
#   you may not use this file except in compliance with the License.
#   You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
#   Unless required by applicable law or agreed to in writing, software
#   distributed under the License is distributed on an "AS IS" BASIS,
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#   See the License for the specific language governing permissions and
#   limitations under the License.
#
from typing import Any

from src.main.python.app.src.app.models.llm.llmprovider import LLMProvider
from src.main.python.app.src.app.models.llm.llmsetting import LLMSetting


class AzureOpenAISetting(LLMSetting):
    def __init__(
        self,
        apiKey,
        model,
        deploymentName,
        privateEndpointBaseUrl,
        apiVersion,
        temperature=None,
        prompt=None,
        **kwargs: Any
    ):
        super().__init__(
            LLMProvider.AZURE_OPEN_AI_SERVICE, apiKey, model, temperature, prompt
        )
        self.deploymentName = deploymentName
        self.privateEndpointBaseUrl = privateEndpointBaseUrl
        self.apiVersion = apiVersion
